<a href="https://colab.research.google.com/github/aimanag/AI-ML-PROJECT/blob/main/AI_ML_DAY_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

In [5]:
df = pd.read_csv('https://talentcocomedia.s3.amazonaws.com/ml-assets/heart-disease.csv')

In [6]:
df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [7]:
df.tail

<bound method NDFrame.tail of      age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   3       145   233    1        0      150      0      2.3   
1     37    1   2       130   250    0        1      187      0      3.5   
2     41    0   1       130   204    0        0      172      0      1.4   
3     56    1   1       120   236    0        1      178      0      0.8   
4     57    0   0       120   354    0        1      163      1      0.6   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  target  
0        0   0     1      

In [9]:
df.shape

(303, 14)


Features in the Above Given DataSet

1.age :- current age of the patient

2.sex :- Gender of the patient (1 refers to Male and 0 refers to Fema3le)

3.cp :- chest pain type --> 1,2,3,0 (essentially of 4 types)

4.trestbps :- The person's resting blood pressure (mm Hg on admission to the hospital) 5.chol :- serum cholestrol in mg/DL

6.fbs :- fasting blood sugar > 120 mg/dl (1 = true; 0 = false)

7.restecg :- resting electrocardiographic results (0 = normal; 1 = having ST-T; 2 = hypertrophy)

8.thalach :- maximum heart rate achieved.

9.exang :- exercise-include angina (True/False)

10.oldpeak :- ST depression induced by exercise relative to rest

11.slope :- The slope of the peak exercise ST segment

12.ca :- number of major vessels (0-3) colored by fluoroscopy

13.thal :- 3 is normal, 6 is fixed defect,7 is reversible defect (thalasemia indicator)

14.target :- the predicted attribute 0 or 1